src/statscollector/DefaultStatsCollector.ts

```diff
@@ -259,6 +262,12 @@ export default class DefaultStatsCollector implements StatsCollector {
           this.clientMetricReport.streamMetricReports[
             Number(rawMetricReport.ssrc)
           ] = streamMetricReport;
+        } else {
+          // Update stream ID in case we have overriden it locally in the case of remote video
+          // updates completed without a negotiation
+          existingStreamMetricReport.streamId = this.videoStreamIndex.streamIdForSSRC(
```

Comment:
When we do stream switch, should we also update the `tileState.streamId`

Reply:
Urgh, this makes me sad how much we exposed `streamId`....added another override function....